rbd/snapshot_remove_by_id.go

```diff
@@ -0,0 +1,25 @@
+//go:build ceph_preview
+// +build ceph_preview
+
+package rbd
+
+/*
+#cgo LDFLAGS: -lrbd
+#include <rbd/librbd.h>
+*/
+import "C"
+
+// RemoveSnapByID removes an existing snapshot. This can be used to manually
+// remove a snapshot from the trash.
+//
+// Implements:
+//
+//	int rbd_snap_remove_by_id(rbd_image_t image, uint64_t snap_id);
+func (image *Image) RemoveSnapByID(snapID uint64) error {
```

Comment:
FWIW when I was filing https://github.com/ceph/go-ceph/issues/1183, I wondered whether go-ceph would try to extend the (IMO completely useless) `Snapshot` abstraction by adding a way to construct it from an ID and then dispatching inside of `Snapshot::Remove()` based on whether it's a name-based or an ID-based instance to call `rbd_snap_remove()` or `rbd_snap_remove_by_id()` respectively or something of that nature.  I'm glad you didn't go there ;)

From the RBD perspective, the more the direct approach of 1:1 mapping to librbd APIs (where there is no `Snapshot`-like entity) is definitely preferred.

Reply:
Yeah, that particular "abstraction" was a bit unfortunate, but it was in place for a while before most of us began contributing and we don't want to break compatibility with other versions. So it is going to continue to exist but it's a good reminder not to rely on it very much for new stuff.